cardinal/router/router.go

```diff
@@ -0,0 +1,238 @@
+package router
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/rotisserie/eris"
+	zerolog "github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"net"
+	"pkg.world.dev/world-engine/cardinal/txpool"
+	shardtypes "pkg.world.dev/world-engine/evm/x/shard/types"
+	routerv1 "pkg.world.dev/world-engine/rift/router/v1"
+	shard "pkg.world.dev/world-engine/rift/shard/v2"
+	"pkg.world.dev/world-engine/sign"
+)
+
+const (
+	defaultPort = "9020"
+)
+
+//go:generate mockgen -source=router.go -package mocks -destination=mocks/router.go
+type Router interface {
```

Comment:
Add a comment on what Router is responsible for

Reply:
added